integrationtests/src/test/java/org/axonframework/integrationtests/testsuite/administration/ImmutableAnnotationBasedAdministrationTest.java

```diff
@@ -0,0 +1,77 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.integrationtests.testsuite.administration;
+
+import org.axonframework.commandhandling.CommandHandlingComponent;
+import org.axonframework.configuration.Configuration;
+import org.axonframework.eventsourcing.EventSourcingRepository;
+import org.axonframework.eventsourcing.eventstore.EventCriteria;
+import org.axonframework.eventsourcing.eventstore.EventStore;
+import org.axonframework.integrationtests.testsuite.administration.common.PersonIdentifier;
+import org.axonframework.integrationtests.testsuite.administration.common.PersonType;
+import org.axonframework.integrationtests.testsuite.administration.state.immutable.ImmutableCustomer;
+import org.axonframework.integrationtests.testsuite.administration.state.immutable.ImmutableEmployee;
+import org.axonframework.integrationtests.testsuite.administration.state.immutable.ImmutablePerson;
+import org.axonframework.messaging.annotation.ParameterResolverFactory;
+import org.axonframework.modelling.annotation.AnnotationBasedEntityIdResolver;
+import org.axonframework.modelling.entity.EntityCommandHandlingComponent;
+import org.axonframework.modelling.entity.EntityModel;
+
+import java.util.ArrayList;
+import java.util.Set;
+
+import static java.lang.String.format;
+
+/**
+ * THIS CLASS ONLY EXIST TO VERIFY THE CURRENT FUNCTIONALITY CAN BE APPLIED WITH ANNOTATIONS. THIS CLASS SHOULD NOT BE
+ * REVIEWED. It's just proof of concept.
+ * <p>
+ * All classes and annotations needed are located in this same class, so it can easily be skipped during review.
+ */
+public class ImmutableAnnotationBasedAdministrationTest extends AbstractAdministrationTestSuite {
+
+    @Override
+    CommandHandlingComponent getCommandHandlingComponent(Configuration configuration) {
+        EntityModel<ImmutablePerson> personModel = new AnnotationTestDefinitions.AnnotatedEventSourcedEntityModel<>(
+                ImmutablePerson.class,
+                configuration.getComponent(ParameterResolverFactory.class),
+                Set.of(ImmutableCustomer.class, ImmutableEmployee.class));
+
+
+        EventSourcingRepository<PersonIdentifier, ImmutablePerson> repository = new EventSourcingRepository<>(
+                PersonIdentifier.class,
+                ImmutablePerson.class,
+                configuration.getComponent(EventStore.class),
+                (type, id) -> {
+                    if (id.type() == PersonType.EMPLOYEE) {
+                        return new ImmutableEmployee(null, null, null, null, null, new ArrayList<>());
```

Comment:
It's interesting... because you already know the id here, but you pass null to do the existence check while processing CreateEmployee command. What should be the recommended way here? 
I think we may use the id here and make it final (even in Mutable case), but add additional property in the class, like `created`. I know it's just a test and it doesn't influence it, but I think we should already think about that how to work with the framework.

Reply:
Honestly, I think this method should only be used if there IS an event in the event stream, otherwise the `loadOrCreate` should be used and pass in a function. 
This way we can achieve nullability. But, we need to find a way to define a creational command handler, that can publish events and immediately create the model based on that. I am experimenting with that locally.

I would suggest that for now we do it like that, but we definitely find a better solution :) It's not very straightforward though, I'm afraid.